### DISCUSSION TITLE
Add additional empty layers

### DISCUSSION THREAD

**Discussion by stickninja** [sn4k3/UVtools#914] [General]
Is it possible to add additional empty layers after the first one? I have been working with another user on the issue of the Phrozen Sonic Mini 8K S and it's compression in the z axis, especially within viscous resins. The issue is, when the printer is initiating a print and it moves the build plate down to start to find home, the z axis rod and motor are flexing under pressure and the 0 setting is thrown off for that print. This happens to every print and can be as much as 1mm. The Mini 8K S also can not have more than 0.5mm z offset applied. I came up with a process to remove the build plate, start a print. After the printer has 'exposed' the first empty layer, I pause the print while it is reading the next layer (can't be before or it will error out). Then I replace the build plate and resume. This totally eliminates the z compression as 0 was found under no tension to the z axis. The problem is, the printer is still ignoring the wait times of the first non-empty layer, i.e. the layer after the empty layer upon resume. I think if we can add an additional empty layer, we can completely mitigate this and get good adhesion and layer thickness as we can use appropriately low bottom exposure times.

**Comment by sn4k3** [comment]
There are multiple ways:

1. You can use the Layer actions to clone the first empty layer.
2. Go to layer 0 and press: `Ctrl + Insert` (Shortcut to 1.)
3. Create a code block and run via terminal at each of your file
4. Create a script and run at each of your file.

But from what you tell, your home point looks not optimal, what about using a thicker sheet to calibrate home? If you have the same trouble with normal resins looks more a home position error.

**Reply by stickninja** [reply]
It's not a matter of leveling. I have tried thicker, and at some point you have 0 adhesion and failure. Basically more than 0.2mm or so (this was Phrozen's rec and also to never print on the build plate).

The printer literally doesn't have any wait time at the bottom of it's travel before setting home, even though the z rod and motor are deflected. Removing any strain on the z axis completely resolves the issue, I am able to level with a single sheet of paper and get as good of z height as I am able to measure with digital calipers.

**Reply by sn4k3** [reply]
Crappy firmwares...
On my printer I have TMC and a [home sequence](https://github.com/sn4k3/klipper-msla-macros/blob/main/optional/tmc_safe_home.cfg) to set a stall guard, as soon the plate hit the plane it stalls and homes, preventing any deflection and force on LCD, goodies of open-source firmware :)

**Reply by stickninja** [reply]
Oh the lamentations. You should see the discussions on the Lychee discord. The new generation of printers fixed 0 issues of the old ones, then added more issues on top of those with the new features such as auto-leveling.

**Reply by sn4k3** [reply]
It's not a new, it's been like that since mSLA is here. They just want to take your money and produce as many junk they can. The false hope of the new product fix the past problems are a selling point for them. Then the cicle repeats because it's profitable. And trust me, is not by any means hard to fix and code a good firmware. They can simply release a patch to fix all problems on all models with just a few code lines which are free and don't require any physical part, being a virtual problem they are in the power to fix or not when they want to.
But the major problem are the costumers who accept and most don't even care as long they can print their toys. The market is ruined most because this people attitude or lack of. If everyone start to return printers, complain and put bad reviews everywhere, they would have other kind of respect and attitude with the clients. Currently it's profitable to make as it is now and that's the problem. Again they do what people allow to.

**Comment by stickninja** [comment]
Well after I posted this I found where I can clone layer 0 (empty)... I will try that first.
